src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -3320,7 +3347,12 @@ def _unary_op_handler(self, node, parent):
         else:
             node_list = [node.items[1]]
         unary_op = UnaryOperation(operator, parent=parent)
-        self.process_nodes(parent=unary_op, nodes=node_list)
+
+        # Store the names of any named args
+        arg_nodes, arg_names = _get_arg_names(node_list)
+        unary_op._named_args = arg_names
```

Comment:
pylint complains of protected access in this and the other named_args setters.

Reply:
This code has been re-written.